robot/data/variables.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/python
+
+##
+#    @file    variables.py
+#    @brief   Contains common IPMI constants
+#
+#    @author  Rahul Maheshwari
+#
+#    @date    June 29, 2016
+##
+
+##
+# @brief This class contains common IPMI constants
```

Comment:
none of these need to be abstracted for IPMI, it's not as if there's another IPMI thing but with different commands.

Instead, abstract away functionality, such as the function of powering on, getting a list of SELs, and be able to have these abstractions run against multiple BMCs, or be sufficiently pluggable that a system with no FSP and no BMC at all would work.